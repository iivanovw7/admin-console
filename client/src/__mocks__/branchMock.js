export const BRANCH = {
  address: '938 Barby Way',
  email: 'mbrazier0@amazon.co.uk',
  fax: '+86 829 425 7497',
  information: '1',
  name: 'San Mart√≠n de los Andes',
  phone: '+54 275 474 9318',
  status: true,
  __v: 0,
  _id: '5c8f96b9cece9911f01dd092'
};