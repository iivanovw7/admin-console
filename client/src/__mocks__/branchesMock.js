export const BRANCHES = {
  page: 1,
  limit: 7,
  pages: 4,
  results: 25,
  output: [
    {
      address: '938 Barby Way',
      email: 'mbrazier0@amazon.co.uk',
      fax: '+86 829 425 7497',
      information: '1',
      name: 'San Mart√≠n de los Andes',
      phone: '+54 275 474 9318',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd092'
    },
    {
      address: '903 Dottie Street',
      email: 'inerger1@businesswire.com',
      fax: '+86 230 609 6364',
      information: 't',
      name: 'Dingzhai',
      phone: '+86 103 802 9655',
      status: true,
      __v: 0,
      _id: '5c8f96b9cece9911f01dd093'
    },
    {
      status:false,
      _id:"5c8f96b9cece9911f01dd0a0",
      name:"Cantao-an",
      email:"atroye@dailymail.co.uk",
      phone:"+63 269 360 0515",
      fax:"+86 713 306 1827",
      address:"7827 Luster Point",
    }
  ]
};
