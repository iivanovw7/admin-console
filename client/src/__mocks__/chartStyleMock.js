export const style = {
  type: "PieChart",
  id: 2
};
