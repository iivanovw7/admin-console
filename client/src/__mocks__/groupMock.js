export const GROUP = {
  status: true,
  permissions: true,
  _id: '5c8f96bacece9911f01dd0a6',
  name: 'Front-End Pool',
  description: 'Group of all employees with Front-End expertise'
};