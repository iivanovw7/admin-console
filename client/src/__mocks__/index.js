export * from './branchesMock';
export * from './branchMock';
export * from './userMock';
export * from './usersMock';
export * from './groupMock';
export * from './groupsMock';
export * from './roleMock';
export * from './rolesMock';
export * from './ticketMock';
export * from './ticketsMock';
export * from './messageMocks';
export * from './messagesMocks';
export * from './statisticsMock';
