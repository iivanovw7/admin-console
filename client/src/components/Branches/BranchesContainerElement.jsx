import { CircularProgress, TableCell, TableRow, Fade } from '@material-ui/core';
import * as axios from 'axios';
import PropTypes from 'prop-types';
import React, { useEffect, useState } from 'react';
import * as URL from '../../constants/api';

export const BranchesContainerElement = props => {

  const { classes, row, handleBranchClick, displayStatus } = props;
  const [users, setUsersQuantity] = useState(0);
  const [isLoading, setIsLoading] = useState(false);
  const [isError, setIsError] = useState(false);
  const queryUrl = `${URL.PRIVATE_API}/stats/branch`;
  const queryParams = { params: { branch: row._id }, withCredentials: true };

  useEffect(() => {
    const countBranchUsers = async () => {
      setIsLoading(true);
      return await axios.get(queryUrl, queryParams)
                        .then(result => {
                          setUsersQuantity(result.data);
                          setIsLoading(false);
                        })
                        .catch(error => {
                          setIsError(true);
                          setIsLoading(false);
                        });
    };
    countBranchUsers();
  }, []);

  return (
    <Fade in={true}>
      <TableRow
        onClick={() => {
          handleBranchClick(row._id);
        }}
        className={classes.branchRowClass}
      >
        <TableCell
          component="th"
          scope="row"
          className={classes.branchNameCell}
        >
          {row.name}
        </TableCell>
        <TableCell className={classes.branchAddressCell} align="center">
          {row.address}
        </TableCell>
        <TableCell className={classes.branchEmployeesCell} align="center">
          {isError && Error}
          {isLoading ? <CircularProgress size={12}/> : users.total}
        </TableCell>
        <TableCell className={classes.tableCell} align="center">
          <strong>
            {displayStatus(row.status)}
          </strong>
        </TableCell>
      </TableRow>
    </Fade>
  );
};

BranchesContainerElement.propTypes = {
  classes: PropTypes.any,
  displayStatus: PropTypes.func.isRequired,
  handleBranchClick: PropTypes.func.isRequired,
  row: PropTypes.object.isRequired
};