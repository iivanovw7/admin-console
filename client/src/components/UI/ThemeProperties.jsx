import blue from '@material-ui/core/colors/blue';
import { fade } from '@material-ui/core/styles/colorManipulator';
import { createMuiTheme } from '@material-ui/core/styles';

const drawerWidth = 240;

//Exporting material-ui main themeProperties configuration
export const themeProperties = createMuiTheme({
  typography: {
    useNextVariants: true
  },
  mixins: {
    toolbar: {
      textAlign: 'center',
      minHeight: '64px',
      justifyContent: 'space-between'
    }
  },
  palette: {
    primary: blue,
    secondary: {
      main: '#ffffff'
    }
  },
  status: {
    danger: 'orange'
  }
});

//Exporting material-ui main styles
export const NavigationStyles = theme => ({
  active: {
    backgroundColor: 'rgba(0, 0, 0, 0.10)'
  },
  root: {
    display: 'flex'
  },
  drawer: {
    [theme.breakpoints.up('sm')]: {
      width: drawerWidth,
      flexShrink: 0
    }
  },
  appBar: {
    marginLeft: drawerWidth,
    [theme.breakpoints.up('sm')]: {
      width: `calc(100% - ${drawerWidth}px)`
    }
  },
  menuButton: {
    marginRight: 20,
    [theme.breakpoints.up('sm')]: {
      display: 'none'
    }
  },
  toolbar: theme.mixins.toolbar,
  toolbarUserName: {
    [theme.breakpoints.down('xs')]: {
      display: 'none'
    }
  },
  drawerPaper: {
    width: drawerWidth
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3
  },
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: fade(theme.palette.common.white, 0.15),
    '&:hover': {
      backgroundColor: fade(theme.palette.common.white, 0.25)
    },
    marginRight: theme.spacing.unit * 2,
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit * 3,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%'
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: 200
    }
  }
});

export const LoginFormStyles = theme => ({

  [theme.breakpoints.down('sm')]: {
    loginFormContainer: {
      boxShadow: 'none',
      height: '100%',
      display: 'flex',
      flexDirection: 'column',
      justifyContent: 'center'
    }
  },
  loginWrapper: {
    maxWidth: '500px',
    margin: '0 auto',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    height: '100%'
  },
  loginFormContainer: {
    padding: theme.spacing.unit,
    maxWidth: '500px'
  },
  formPaper: {
    margin: theme.spacing.unit * 2
  },
  paddingTop: theme.spacing.unit * 2,
  paddingBottom: theme.spacing.unit * 2

});

//Exporting material-ui main styles
export const Wrapper = theme => ({
  root: {
    display: 'flex'
  },
  toolbar: theme.mixins.toolbar,
  contentSingle: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  contentList: {
    flexGrow: 1,
    padding: theme.spacing.unit * 3,
    marginTop: '64px',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'space-between',
    [theme.breakpoints.up('sm')]: {
      height: '90vh'
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    }
  },
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap',

    buttonWrapper: {
      display: 'flex',
      alignItems: 'center'
    }
  },
  fromLimitSelector: {
    display: 'flex',
    flexWrap: 'wrap',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  formControl: {
    margin: theme.spacing.unit,
    minWidth: 120,
    maxWidth: 300
  },
  controlsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    paddingRight: 10,
    paddingLeft: 10
  },
  selectorsContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    paddingRight: 10,
    paddingLeft: 10
  }
});

export const Container = theme => ({
  root: {
    width: '100%',
    marginTop: theme.spacing.unit * 3,
    overflowX: 'auto',
    display: 'flex',
    justifyContent: 'center',
    overflow: 'hidden'
  },
  tables: {
    minWidth: 200,
    width: '100%'
  },
  tableCell: {
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5
  },
  rowClass: {
    '&:hover': {
      backgroundColor: fade(theme.palette.common.black, 0.1),
      cursor: 'pointer'
    },
    [theme.breakpoints.up('sm')]: {
      height: '48px',
    },
    [theme.breakpoints.down('sm')]: {
      height: '100%',
    },
  },
  nameCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,

    [theme.breakpoints.up('sm')]: {
      width: '30%',
    },

    [theme.breakpoints.down('sm')]: {
      width: '60%',
    }

  },
  addressCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '100%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  employeesCell: {
    paddingRight: 5,
    paddingLeft: 10,
    paddingTop: 5,
    paddingBottom: 5,
    width: '30%',
    [theme.breakpoints.down('sm')]: {
      display: 'none'
    }
  },
  branchPaper: {
    width: '100%',
    maxWidth: 600,
    margin: theme.spacing.unit * 2
  }
});

export const Notification = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  }
});

export const Buttons = theme => ({
  formRoot: {
    display: 'flex',
    flexWrap: 'wrap'
  },
  [theme.breakpoints.down('sm')]: {
    mobile: {
      display: 'flex'
    },
    desktop: {
      display: 'none'
    }
  },
  [theme.breakpoints.up('sm')]: {
    mobile: {
      display: 'none'
    },
    desktop: {
      display: 'flex'
    }
  }
});

