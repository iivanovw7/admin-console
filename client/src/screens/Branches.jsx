import React, { useState, useEffect } from 'react';
import { Helmet } from 'react-helmet';
import PropTypes from 'prop-types';
import { CssBaseline, Paper } from '@material-ui/core';
import AppBarContainer from '../components/UI/AppBar/AppBarContainer';
import DrawerContainer from '../components/UI/Drawer/DrawerContainer';
import { withStyles } from '@material-ui/core/styles';
import { Wrapper } from '../components/UI/ThemeProperties';
import BranchesListContainer from '../components/Branches/BranchesContainer';
import { PageSelector } from '../components/UI/PageSelector';
import AddNewButton from '../components/UI/AddButton';
import { getBranches } from '../actions/branches';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

const Branches = props => {
  const { classes, history } = props;
  const limit = 8; //limit of elements for current page

  //current page number
  const [currentPage, setCurrentPage] = useState(props.branches.list.page);
  const list = props.branches.list.output; //list of elements fetched

  useEffect(() => {
    props.dispatch(getBranches(currentPage, limit, history));
  }, [currentPage]);


  function handlePage(newPage) {
    setCurrentPage(newPage);
  }

  return (
    <main className={classes.contentList}>
      <div>
        <Paper className={classes.controlsContainer}>
          <div className={classes.selectorsContainer}>
            <h2>Branches</h2>
          </div>
          <AddNewButton history={history} element={'branches'}/>
        </Paper>
        {
          (list) ?
            (<BranchesListContainer dispatch={props.dispatch}/>) : (<p>Loading...</p>)
        }
      </div>
      <br/>
      <Paper className={classes.controlsContainer}>
        <div/>
        <PageSelector classes={classes} data={props.branches} handlePage={handlePage}/>
      </Paper>
    </main>
  );
};

Branches.propTypes = {
  classes: PropTypes.object.isRequired,
  theme: PropTypes.object.isRequired
};

function mapStateToProps(state) {
  return { branches: state.branches };
}

export default connect(mapStateToProps, { getBranches })(withStyles(Wrapper, { withTheme: true })(withRouter(Branches)));

