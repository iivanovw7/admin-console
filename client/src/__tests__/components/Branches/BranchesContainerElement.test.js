import { Table, TableBody, TableHead, withStyles } from '@material-ui/core';
import { createMount, createShallow } from '@material-ui/core/test-utils';
import { createMemoryHistory } from 'history';
import jest from 'jest-mock';
import React from 'react';
import * as mocks from '../../../__mocks__/';
import { BranchesContainerElement } from '../../../components/Branches/BranchesContainerElement';
import { Container } from '../../../components/UI/ThemeProperties';

const setup = () => {
  let wrapper;
  let shallowWithDive;
  let mount;
  let shallow;
  const history = createMemoryHistory('/dashboard');

  const Composition = withStyles(Container, { withTheme: true })(BranchesContainerElement);

  const props = {
    row: mocks.BRANCH,
    history: history
  };

  beforeEach(() => {
    shallow = createShallow();
    shallowWithDive = createShallow({ dive: true });
    mount = createMount();
  });

  afterEach(() => {
    mount.cleanUp();
  });

  /**
   *  Renders single branch container with styles
   */
  describe('Branch container element', () => {
    it('Should render branch container element with fields and styles', () => {
      const mockBranchClick = jest.fn();
      const mockDisplayStatus = jest.fn();
      wrapper = mount(
        <Table>
          <TableHead/>
          <TableBody>
            <Composition
              {...props}
              handleBranchClick={mockBranchClick}
              displayStatus={mockDisplayStatus}
            />
          </TableBody>
        </Table>
      );
      expect(wrapper.html()).not.toBe(null);
      expect(wrapper.length).toEqual(1);
      expect(wrapper.html()).toMatchSnapshot();

      //checking branch click event handling
      wrapper.find(BranchesContainerElement).simulate('click');
      expect(mockDisplayStatus).toHaveBeenCalled();
      expect(mockBranchClick).toHaveBeenCalled();
      expect(mockBranchClick).toHaveBeenCalledTimes(1);

    });
  });
};

setup();
