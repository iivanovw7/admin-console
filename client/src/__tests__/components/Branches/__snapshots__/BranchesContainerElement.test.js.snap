// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branch container element Should render branch container element with fields and styles 1`] = `"<table class=\\"MuiTable-root-1\\"><thead class=\\"MuiTableHead-root-2\\"></thead><tbody class=\\"MuiTableBody-root-3\\"><tr class=\\"MuiTableRow-root-30 BranchesContainerElement-branchRowClass-12\\" style=\\"opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"><th class=\\"MuiTableCell-root-35 MuiTableCell-body-37 BranchesContainerElement-branchNameCell-13\\" scope=\\"row\\">San Mart√≠n de los Andes</th><td class=\\"MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44 BranchesContainerElement-branchAddressCell-14\\">938 Barby Way</td><td class=\\"MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44 BranchesContainerElement-branchEmployeesCell-15\\"><div class=\\"MuiCircularProgress-root-47 MuiCircularProgress-colorPrimary-50 MuiCircularProgress-indeterminate-49\\" style=\\"width: 12px; height: 12px;\\" role=\\"progressbar\\"><svg class=\\"MuiCircularProgress-svg-52\\" viewBox=\\"22 22 44 44\\"><circle class=\\"MuiCircularProgress-circle-53 MuiCircularProgress-circleIndeterminate-55\\" cx=\\"44\\" cy=\\"44\\" r=\\"20.2\\" fill=\\"none\\" stroke-width=\\"3.6\\"></circle></svg></div></td><td class=\\"MuiTableCell-root-35 MuiTableCell-body-37 MuiTableCell-alignCenter-44 BranchesContainerElement-tableCell-11\\"><strong></strong></td></tr></tbody></table>"`;
